app/src/main/java/org/eyeseetea/malariacare/data/remote/PushDhisSDKDataSource.java

```diff
@@ -0,0 +1,112 @@
+import android.net.NetworkInfo;
+import android.util.Log;
+
+import com.raizlabs.android.dbflow.sql.language.Delete;
+
+import org.eyeseetea.malariacare.data.IDataSourceCallback;
+import org.eyeseetea.malariacare.data.database.utils.PreferencesState;
+import org.eyeseetea.malariacare.domain.exception.NetworkException;
+import org.hisp.dhis.client.sdk.android.api.D2;
+import org.hisp.dhis.client.sdk.android.api.persistence.flow.EventFlow;
+import org.hisp.dhis.client.sdk.android.api.persistence.flow.StateFlow;
+import org.hisp.dhis.client.sdk.android.api.persistence.flow.TrackedEntityDataValueFlow;
+import org.hisp.dhis.client.sdk.models.common.importsummary.ImportSummary;
+import org.hisp.dhis.client.sdk.models.event.Event;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import rx.Observable;
+import rx.android.schedulers.AndroidSchedulers;
+import rx.functions.Action1;
+import rx.schedulers.Schedulers;
+
+public class PushDhisSDKDataSource {
+
+    public void pushData(final IDataSourceCallback<Map<String, ImportSummary>> callback) {
+        boolean isNetworkAvailable = isNetworkAvailable();
+
+        if (!isNetworkAvailable) {
+            callback.onError(new NetworkException());
+        } else {
+            pushEvents(callback);
+        }
+    }
+
+    private void pushEvents(final IDataSourceCallback<Map<String, ImportSummary>> callback) {
+        final Set<String> eventUids= new HashSet<>();
+
+        for(EventFlow eventFlow:SdkQueries.getEvents()){
+                eventUids.add(eventFlow.getUId());
+           }
+        Observable<Map<String,ImportSummary>> eventObserver =
+                       D2.events().push(eventUids);
+
+        eventObserver
+        .subscribeOn(Schedulers.io())
+        .observeOn(AndroidSchedulers.mainThread())
+        .subscribe(new Action1<Map<String, ImportSummary>>() {
+            @Override
+            public void call(Map<String, ImportSummary> mapEventsImportSummary) {
+                Log.d(this.getClass().getSimpleName(),
+                        "Push of events finish. Number of events: "
+                                + mapEventsImportSummary.size());
+
+                //TODO: from data source should comverto always from SDK object to domain object
+                // this class should not return sdk objects directly
+                //create a object similar to Map<String,ImportSummary> in domain and convert before
+                // to invoke callback.onSuccess
+                callback.onSuccess(mapEventsImportSummary);
+            }
+        }, new Action1<Throwable>() {
+            @Override
+            public void call(Throwable throwable) {
+
+                callback.onError(throwable);
+
+                Log.e(this.getClass().getSimpleName(),
+                        "Error pushing Events: " + throwable.getLocalizedMessage());
+            }
+        });
+    }
+
+    private boolean isNetworkAvailable() {
```

Comment:
Is a method without logic, simply invoked getActiveNetworkInfo from ConnectivityManager. I think not is problem this duplicate code